pyro/contrib/epidemiology/compartmental.py

```diff
@@ -868,3 +849,77 @@ def init(self, state):
     def step(self, state):
         with poutine.block(hide_types=["observe"]):
             super().step(state.copy())
+
+
+class _HaarSplitReparam:
+    """
+    Wrapper around ``HaarReparam`` and ``SplitReparam`` to additionally convert
+    sample dicts between user-facing and auxiliary coordinates.
+    """
+    def __init__(self, split, duration, dims, supports):
+        assert 0 <= split < duration
+        self.split = split
+        self.duration = duration
+        self.dims = dims
+        self.supports = supports
+
+    def __bool__(self):
+        return True
+
+    def reparam(self, model):
+        """
+        Wrap a model with ``poutine.reparam``.
+        """
+        # Transform to Haar coordinates.
+        config = {}
+        for name, dim in self.dims.items():
+            config[name] = HaarReparam(dim=dim, flip=True)
```

Comment:
make `flip` an arg?

Reply:
I think that's unnecessary since we don't expose any plumbing to set that, and since `_HaarSplitReparam` is merely a minimal internal helper.